src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -1010,8 +1012,19 @@ class RandomStringFromDataset : public Generator<std::string> {
 
     std::string evaluate() {
         auto dataset = _datasets.getDatasetForPath(_path);
-        auto distribution = boost::random::uniform_int_distribution<size_t>{0, dataset.size() - 1};
-        return dataset[distribution(_rng)];
+        if(_sequential) {
+            // We can't check this until evaluation time, but this can only happen when startFromLine > # lines in dataset.
```

Comment:
Why can't we check this after loading the data in the constructor?

Reply:
Let me look into this more -- it will probably require a redesign of the DatasetCache.

Reply:
Changed this by redesigning the cache, let me know what you think.